src/search/indexer.cc

```diff
@@ -269,6 +266,32 @@ Status IndexUpdater::UpdateNumericIndex(std::string_view key, const kqir::Value
   return Status::OK();
 }
 
+Status IndexUpdater::UpdateHnswVectorIndex(std::string_view key, const kqir::Value &original,
+                                           const kqir::Value &current, const SearchKey &search_key,
+                                           HnswVectorFieldMetadata *vector) const {
+  CHECK(original.IsNull() || original.Is<kqir::NumericArray>());
+  CHECK(current.IsNull() || current.Is<kqir::NumericArray>());
+
+  auto storage = indexer->storage;
+  auto hnsw = HnswIndex(search_key, vector, storage);
```

Comment:
HnswIndex is a relatively heavy class (since it includes `mt19937`, `sizeof(mt19937) ~= 5000`), so I'm wondering if we can optimize to avoid constructing it every time.

Reply:
Ahh I wasn't aware of that, and thanks for pointing it out!

One way I could think of to avoid heavy construction is to make all instances of HnswIndex share a single `mt19937` instance. 

We can implement a `RandomGenerator` using Singleton Pattern as below:

```
class RandomGenerator {
public:
    static RandomGenerator& getInstance() {
        static RandomGenerator instance;
        return instance;
    }

    std::mt19937& getGenerator() {
        return generator_;
    }

private:
    RandomGenerator() {
        std::random_device rd;
        generator_ = std::mt19937(rd());
    }
    // delete copy & move constructor
    std::mt19937 generator_;
};

struct HnswIndex {
    HnswIndex(...) {
        generator_ = &RandomGenerator::getInstance().getGenerator();
    }
};
```

I'll open an issue for this, and see if anyone has better solutions. :)

Reply:
Yeah we can make it a static member.

For static local variables I don't prefer them because they will introduce some mutexes.

Reply:
Gotta, then we could potentially have something like
```
#include <random>

class HnswIndex {
// ...
private:
    // ...
    static std::mt19937 generator_;  // Static member variable
};

std::mt19937 HnswIndex::generator_{std::random_device{}()};
```